### PR TITLE
[JBTM-2640] Add username and password to JmsXAResourceRecoveryHelper

### PR DESCRIPTION
**Pull request by gytis**
https://issues.jboss.org/browse/JBTM-2640

!BLACKTIE !PERF !XTS !QA_JTA !QA_JTS_JACORB !QA_JTS_JDKORB !QA_JTS_IBMORB

### CODE REVIEW

**Comment by gytis** [comment]
WildFly rebase failed. But JMS tests have passed.

File: ArjunaJTA/jms/src/test/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelperTests.java [hunk 1]
```diff
@@ -83,6 +84,15 @@ public void shouldCreateConnectionOnScanStart() throws Exception {
     }
 
     @Test
+    public void shouldCreateConnectionWithCredentialsOnScanStart() throws Exception {
+        when(xaConnectionFactoryMock.createXAConnection(anyString(), anyString())).thenReturn(xaConnectionMock);
```
Comment (mmusgrov) [inline ArjunaJTA/jms/src/test/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelperTests.java]: That's horrible but I guess it's the "Mockito Way"

Comment (gytis) [inline ArjunaJTA/jms/src/test/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelperTests.java]: Why is that horrible? I don't know any easier way to mimic such scenario. Just telling the connection factory to return a specific connection instance, if that method is called with any string arguments.

Comment (mmusgrov) [inline ArjunaJTA/jms/src/test/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelperTests.java]: It was just a remark about the way Mockito does things. 

    if (xaConnectionFactoryMock.createXAConnection(anyString(), anyString()) {
        return(xaConnectionMock);
    }

is the way to go IMHO

Comment (mmusgrov) [inline ArjunaJTA/jms/src/test/java/org/jboss/narayana/jta/jms/JmsXAResourceRecoveryHelperTests.java]: Ah wait I think I am misunderstanding the Mockito APIs.

**Pull request merged**